AUTO-3610: add support for HTTP API (#34)
pepsico-ecommerce/snowflex: README.md

[![License](https://img.shields.io/hexpm/l/snowflex.svg)](https://github.com/pepsico-ecommerce/snowflex/blob/master/LICENSE)
[![Last Updated](https://img.shields.io/github/last-commit/pepsico-ecommerce/snowflex.svg)](https://github.com/pepsico-ecommerce/snowflex/commits/master)

This application includes a functional interface, as well as a `:db_connection`
driver for connecting to a Snowflake data warehouse. By default, this is backed
by an `:odbc` connection pool, but it may also use Snowflake's HTTP API.

## ODBC Setup

For the `:odbc` transport, make sure your Erlang runtime includes the `:odbc`
application. The following config options can be set:

```elixir
config :snowflex,
  transport: :odbc, # not required, since this is the default transport
  driver: "/path/to/my/ODBC/driver" # defaults to "/usr/lib/snowflake/odbc/lib/libSnowflake.so"
```

> For the `:http` transport, you'll also need to install the optional `:tesla`
> and `:jason` dependencies, along with a Tesla-compatible network adapter
> (`:hackney` is recommended):

Reviewer: ```suggestion
For the `:http` transport, you'll also need to install the optional `:req`
and `:jason` dependencies.
```

```elixir
# config.exs
config :snowflex, transport: :http

> config :tesla, adapter: Tesla.Adapter.Hackney

Reviewer: ```suggestion
```

Not sure what this should be really

# your_app/mix.exs
def deps do
  [
    # ...

>     {:tesla, "~> 1.4"},
>     {:jason, "~> 1.3"},
>     {:hackney, "~> 1.18"}

Reviewer: ```suggestion
    {:req, "~> 0.5.0"},
    {:jason, "~> 1.3"},
```
